StepCounterKotlin/app/src/main/java/com/google/android/gms/fit/samples/stepcounter/MainActivity.kt

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.fit.samples.stepcounter
+
+import android.Manifest
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Color
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.Settings
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.widget.TextViewCompat
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.fit.samples.common.logger.Log
+import com.google.android.gms.fit.samples.common.logger.LogView
+import com.google.android.gms.fit.samples.common.logger.LogWrapper
+import com.google.android.gms.fit.samples.common.logger.MessageOnlyLogFilter
+import com.google.android.gms.fitness.Fitness
+import com.google.android.gms.fitness.FitnessOptions
+import com.google.android.gms.fitness.data.DataType
+import com.google.android.gms.fitness.data.Field
+import com.google.android.material.snackbar.Snackbar
+
+const val TAG = "StepCounter"
+
+/**
+ * This sample demonstrates combining the Recording API and History API of the Google Fit platform
+ * to record steps, and display the daily current step count. It also demonstrates how to
+ * authenticate a user with Google Play Services.
+ */
+class MainActivity : AppCompatActivity() {
+    private val fitnessOptions = FitnessOptions.builder()
+            .addDataType(DataType.TYPE_STEP_COUNT_CUMULATIVE)
+            .addDataType(DataType.TYPE_STEP_COUNT_DELTA)
+            .build()
+
+    // Two maps are used in conjunction with the OAuth flow. This is in order to identify what
+    // function to execute once the flow has successfully completed.
+    private val signInActionMap = mapOf<() -> Any, Int>(
+            ::subscribe to 1,
```

Comment:
I would make constants with Request Code in the name so it is clear from the start the you use it for that.

Reply:
It seems hard to make these constants using the function reference technique, so I've instead extracted them to an enum at the top level. On the negative side, this means having to manually map each code to each function for post-sign-in execution, in `performActionForRequestCode`, but on the plus side, I think it is more intuitive now. wdyt?

Reply:
SGTM!

BTW, if you really prefer the original approach, you could also add comments above explaining that each int represents a request code too.

Reply:
Actually, I really like the new version. Feels easier to read, so I would stick with that. :)